### ISSUE TITLE
Fhir client for Angular2 

### ISSUE THREAD

**Issue by bunyaminsg** [FHIR/fhir.js#90]
Since fhir.js doesn't have an Angular2 adapter, I created an [npm package](https://www.npmjs.com/package/ng-fhir) and an [example project](https://github.com/bunyaminsg/ng-fhir-example) to use fhir.js with Angular2. I would be glad if anyone interested in Angular could contribute these projects to make them better and more usable.

**Comment by niquola** [comment]
It would be nice if you will pull-request support for ng-2 :) Is it really different from ng-1 in terms of xhr?

**Comment by bunyaminsg** [comment]
I didn't make any changes in terms xhr, actually I didn't make any changes on the source of fhir.js :) I only used the native build of fhir.js, and created a class that use directly the methods of the native client. The only difference is that my class has a declaration file to make it importable for Angular2. 

Since I worked on the build, not on the source, I don't know where to put my code on your repository. If you can decide how to integrate it with your project, I could do a pull-request.

**Comment by syamraj** [comment]
Thank you bunyaminsg for your effort in coming up with the package. But i run into error when using FhirClient. Cause of the error is "import { fhir } from './nativeFHIR';" in the FhirClient.d.ts. Error says --allow.js is not set. After setting --allow.js to true in tsconfig.json it hit one more error "has no exported member 'fhir'". Help would be really appreciated :) . By the way I have my project build with "@angular/cli": "1.0.1".

**Comment by bunyaminsg** [comment]
@syamraj I will appreciate to help you about this issue but can you write it to [ng-fhir](https://github.com/bunyaminsg/ng-fhir/issues) repository? It should be better to not keep busy fhir.js repository by the issues related with ng-fhir.

**Comment by emreavsar** [comment]
hey @niquola  and @bunyaminsg any updates here on a angular adapter for fhir? I'm currently using ng-fhir but I just found out that it does not return any headers in the response, and I need them.

@bunyaminsg i think it is not your task to adjust the ng-fhir (as far as I understood it is just a wrapper of the dist of fhir.js), it should be kind of a 'official' fhir-js adapter for angular (not angularjs), nah?

**Comment by bunyaminsg** [comment]
@emreavsar yes, it is just a wrapper to use fhir.js client functions on angular. I'm using it by changing the dist of fhir.js according to my requirements (e.g. for custom headers). It would be great to have a more advanced, configurable and type safe angular library. I'll appreciate to contribute if there will be a work on this.
